src/helpers.rs

```diff
@@ -348,4 +349,42 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         Ok(())
     }
+
+    fn read_os_string(&mut self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString> {
+        let bytes = self.eval_context_mut().memory().read_c_str(scalar)?.to_vec();
+        if cfg!(unix) {
+            Ok(std::os::unix::ffi::OsStringExt::from_vec(bytes))
```

Comment:
This will fail to compile on Windows. You need `#[cfg(unix)]`, not just `if`.

Reply:
My bad, I'm unexperienced with platform specific code. Let me fix it.